pkg/trace/sampler/signature.go

```diff
@@ -91,3 +90,34 @@ func computeSpanHash(span *pb.Span, env string, withResource bool) spanHash {
 	}
 	return spanHash(h.Sum32())
 }
+
+// sum32a is an adaptation of https://golang.org/pkg/hash/fnv/#New32a, but simplified
+// for our use case to remove interfaces which caused unnecessary allocations.
+type sum32a uint32
+
+const (
+	offset32 = 2166136261
+	prime32  = 16777619
+)
+
+func new32a() sum32a {
+	return offset32
+}
+
+func (s *sum32a) Write(data []byte) {
+	hash := *s
+	for _, c := range data {
+		hash ^= sum32a(c)
+		hash *= prime32
+	}
+	*s = hash
+}
+
+func (s *sum32a) WriteByte(c byte) {
```

Comment:
An odd one here in [circleci/unit_tests](https://app.circleci.com/pipelines/github/DataDog/datadog-agent/10233/workflows/188626d5-ce00-4e70-b9c8-22515010ad63/jobs/300993):
```
# github.com/DataDog/datadog-agent/pkg/trace/sampler
pkg/trace/sampler/signature.go:116:18: method WriteByte(c byte) should have signature WriteByte(byte) error
```
I guess it's trying to force you to implement [`io.ByteWriter`](https://golang.org/pkg/io/#ByteWriter). I'll leave it up to you to decide. If you want to change the name back, or just return `nil`, fine by me.

Reply:
That's an odd linter error indeed... I'll just revert to the previous name, there is no need to add an error argument here.